src/libdrakvuf/win-processes.c

```diff
@@ -861,3 +861,102 @@ bool win_get_process_data( drakvuf_t drakvuf, addr_t base_addr, proc_data_priv_t
 
     return false;
 }
+
+status_t win_find_mmvad(drakvuf_t drakvuf, addr_t eprocess, addr_t vaddr, mmvad_info_t *out_mmvad)
+{
+    addr_t node_addr = eprocess + drakvuf->offsets[EPROCESS_VADROOT] + drakvuf->offsets[VADROOT_BALANCED_ROOT];
+
+    while (node_addr) {
```

Comment:
I wonder if we should have a sanity check on this loop in case there is some recursion because of an error. Say limit how many iterations it should do at max?

Reply:
As this is some kind of AVL tree, in fact it should be very uncommon to have height higher than several dozens. I bet a random constant of 100 should work in normal situations, going to add this sanity check.